src/node.rs

```diff
@@ -912,64 +899,74 @@ impl KademliaNode {
 
         info!("Buckets refreshed");
         Ok(())
+
     }
 
     pub async fn find_node(
-        &self,
+        self: &Arc<Self>,
```

Comment:
If the motivation here is to allow for something like the below

```rust
async fn something() {
  let node = KademliaNode::new();

  let node_clone_for_bg_task = node.clone();
  tokio::spawn(async move {
    node_clone_for_bg_task.find_node_rpc(...).await;
  });
}
```

Then we can just do 

```suggestion
        &self,
```

And call (a custom) `.clone()` before moving the `KademliaNode` object into the separate task.